apps/hash-frontend/src/pages/[shortname].page/edit-user-profile-info-modal/user-profile-info-modal-header.tsx

```diff
@@ -0,0 +1,394 @@
+        userProfile.hasCoverImage.imageEntity.properties,
+      )
+    : undefined;
+
+  const avatarSrc = userProfile.hasAvatar
+    ? getImageUrlFromEntityProperties(
+        userProfile.hasAvatar.imageEntity.properties,
+      )
+    : undefined;
+
+  const handleChangeAvatarImage = useCallback(
+    () => avatarImageInputRef.current?.click(),
+    [],
+  );
+
+  const handleChangeCoverImage = useCallback(
+    () => coverImageInputRef.current?.click(),
+    [],
+  );
+
+  const handleAvatarImageFileUpload = useCallback<
+    ChangeEventHandler<HTMLInputElement>
+  >(
+    async (event) => {
+      const file = event.target.files?.[0];
+
+      if (!file) {
+        throw new Error("No file provided");
+      }
+
+      setNewAvatarImageUploading(true);
+
+      // Upload the file and get a file entity which describes it
+      const { data: fileUploadData, errors: fileUploadErrors } =
+        await uploadFile({
+          data: {
+            description: `The avatar for the ${userProfile.preferredName} user in HASH`,
+            name: `${userProfile.preferredName}'s avatar image`,
+            file,
+            ...(existingAvatarImageEntity
+              ? {
+                  fileEntityUpdateInput: {
+                    existingFileEntityId: existingAvatarImageEntity.metadata
+                      .recordId.entityId as EntityId,
+                  },
+                }
+              : {
+                  fileEntityCreationInput: {
+                    entityTypeId: types.entityType.imageFile.entityTypeId,
+                  },
+                }),
+          },
+        });
+
+      if (fileUploadErrors || !fileUploadData) {
+        throw new Error(
+          fileUploadErrors?.[0]?.message ?? "Unknown error uploading file",
+        );
+      }
+
+      if (userProfile.hasAvatar) {
+        // Delete the existing hasAvatar link, if any
+        await archiveEntity({
+          data: {
+            entityId:
+              userProfile.hasAvatar.linkEntity.metadata.recordId.entityId,
+          },
+        });
+      }
+
+      // Create a new hasAvatar link from the org to the new file entity
+      await createEntity({
+        data: {
+          entityTypeId: types.linkEntityType.hasAvatar.linkEntityTypeId,
+          linkData: {
+            leftEntityId: userProfile.entityRecordId.entityId,
+            rightEntityId: fileUploadData.metadata.recordId
+              .entityId as EntityId,
+          },
+          properties: {},
+        },
+      });
+
+      /** @todo: error handling */
+
+      void refetchUserProfile();
+      void refetchUserAndOrgs();
+
+      setNewAvatarImageUploading(false);
+    },
+    [
+      archiveEntity,
+      createEntity,
+      existingAvatarImageEntity,
+      refetchUserAndOrgs,
+      uploadFile,
+      userProfile,
+      refetchUserProfile,
+    ],
+  );
+
+  const handleCoverImageFileUpload = useCallback<
+    ChangeEventHandler<HTMLInputElement>
+  >(
+    async (event) => {
```

Comment:
Wasn't super trivial because in the `orgForm` the handling is a bit different because the org may or may not be created yet, but have extracted the functionality into a hook which will be useful for me at a later point in https://github.com/hashintel/hash/pull/3338/commits/5da3733fea8642617f4bc34f9905f4b5d98e6ed2